### PR TITLE
DistroInfo source

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following encapsulates the distro info in a type that can be exposed
to juju directly. The series package defines functions at a package
level and it makes it hard to extract the data we need from with in
juju.

The code in this commit takes baby steps towards moving the code into a
type so we can access that from inside juju. It also provides backwards
compatibility to ensure that we don't break juju releases.

Interestingly, the series package is time sensitive (see
updateLocalSeriesVersions), where we don't pass a clock in to be able to
support fixing time to one spot.

### CODE REVIEW

File: series/supportedseries.go [hunk 1]
```diff
@@ -378,6 +378,10 @@ func GetOSFromSeries(series string) (os.OSType, error) {
 	if err == nil {
 		return osType, nil
 	}
+
+	seriesVersionsMutex.Lock()
+	defer seriesVersionsMutex.Unlock()
```
Comment (SimonRichardson) [inline series/supportedseries.go]: This was found with -race.

File: series/supportedseries_linux_test.go [hunk 2]
```diff
@@ -102,9 +102,9 @@ func (s *supportedSeriesSuite) TestSupportedJujuControllerSeries(c *gc.C) {
 	filename := filepath.Join(d, "ubuntu.csv")
 	err := ioutil.WriteFile(filename, []byte(distInfoData), 0644)
 	c.Assert(err, jc.ErrorIsNil)
-	s.PatchValue(series.DistroInfo, filename)
+	s.PatchValue(series.UbuntuDistroInfoPath, filename)
 
-	expectedSeries := []string{"bionic", "disco", "eoan", "xenial"}
```
Comment (SimonRichardson) [inline series/supportedseries_linux_test.go]: We should patch the `time.Now()` in `updateLocalSeriesVersions` as the package goes out of date with every passing day. The tests expect dates to be at a certain time.

**Comment by howbazaar** [comment]
go fmt is sad: series/series_linux.go

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
https://trello.com/c/5td5Xpjn/2415-os-series-introduce-types-to-help-aid-with-testing-and-reasoning

**Pull request merged**